feat: Added docs for multi-platform plugin hook support (#1627)
helm/helm-www: content/en/docs/topics/plugins.md

`--foo` is silently discarded.

### The `platformCommand` Field

The `platformCommand` configures the command that the plugin will execute when
it is called. The following rules will apply in deciding which command to use:

- If `platformCommand` is present, it will be searched first.
  - If both `os` and `arch` match the current platform, search will stop and the
  command will be used.
  - If `os` matches and `arch` is empty, the command will be used.
  - If `os` and `arch` are empty, the command will be used.
- If no `platformCommand` match is found and the deprecated `command` is present
  it will be used.
- If no matches are found in `platformCommand` and no `command` is present, Helm
  will exit with an error.

## Building a Plugin

Here is the plugin YAML for a simple plugin that helps get the last release name:

```yaml
name: last
version: 0.1.0
usage: get the last release name
description: get the last release name
ignoreFlags: false
platformCommand:
  - command: ${HELM_BIN}
    args:
      - list
      - --short
      - --max=1
      - --date
      - -r
```

Plugins may require additional scripts and executables.

> Scripts can be included in the plugin directory and executables can be downloading

Reviewer: ```suggestion
Scripts can be included in the plugin directory and executables can be downloaded
```

> Plugins may require additional scripts, which can be included in the plugin directory,
> or executables, which require downloading via a hook. An example of the directory
> structure and plugin YAML for plugin based on an executable is below:

Reviewer: ```suggestion
Plugins may require additional scripts and executables.
Scripts can be included in the plugin directory and executables can be downloading via a hook.
The following is an example plugin:
```

```console
$HELM_PLUGINS/
  |- myplugin/
    |- scripts/
      |- install.ps1
      |- install.sh
    |- plugin.yaml
```

```yaml
name: myplugin
version: 0.1.0
usage: example plugin
description: example plugin
ignoreFlags: false
platformCommand:
  - command: ${HELM_PLUGIN_DIR}/bin/myplugin
  - os: windows

>     command: ${HELM_PLUGIN_DIR}\bin\myplugin.exe

Reviewer: Does yaml handle these `\` or does it help to quote the strings here? I'm not a windows person.

Author: Strings in YAML rarely need quoting and `\` isn't an escape character in YAML strings.

  `--help`. Helm will use `usage` and `description` for `helm help` and `helm
  help myplugin`, but will not handle `helm myplugin --help`.

### Testing a Local Plugin

First you need to find your `HELM_PLUGINS` path to do it run the folowing command:

``` bash
helm env
```

Change your current directory to the director that `HELM_PLUGINS` is set to.

> Now you can add a symbolic link to your build out put of your plugin in this example

Reviewer: ```suggestion
Now you can add a symbolic link to your build output of your plugin in this example
```

Author: Good catch, I just moved this block but I should have checked it.
